### PR TITLE
Flannel add-on

### PR DESCRIPTION
**Pull request by emosbaugh**
#### What this PR does / why we need it:

Adds Flannel CNI add-on.

#### Which issue(s) this PR fixes:

Fixes NONE

#### Special notes for your reviewer:

https://github.com/replicatedhq/kURL-api/pull/48
https://github.com/replicatedhq/kurl.sh/pull/894
https://github.com/replicatedhq/kots/pull/3337

#### Does this PR introduce a user-facing change?

```release-note
Adds [Flannel add-on](https://kurl.sh/docs/add-ons/flannel) version 0.20.0
```

#### Does this PR require documentation?

https://github.com/replicatedhq/kurl.sh/pull/894

### CODE REVIEW

File: addons/flannel/template/base/yaml/troubleshoot.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: flannel-troubleshoot-spec
+  labels:
+    troubleshoot.io/kind: supportbundle-spec
```
Comment (diamonwiggins) [inline addons/flannel/template/base/yaml/troubleshoot.yaml:6]: should be `support-bundle-spec`

**Review by diamonwiggins — APPROVED** [review]
lgtm

**Review by laverya — APPROVED** [review]

**Comment by laverya** [comment]
Could we get an on-merge and daily test for this?

File: addons/flannel/template/base/install.sh [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+export POD_CIDR
+export POD_CIDR_RANGE
+export POD_CIDR_IPV6
+export EXISTING_POD_CIDR
+
+export FLANNEL_ENABLE_IPV4=true
+export FLANNEL_ENABLE_IPV6=false # TODO: support ipv6
+export FLANNEL_BACKEND=vxlan # TODO: support encryption
+
+function flannel_pre_init() {
```
Comment (emosbaugh) [inline addons/flannel/template/base/install.sh:12]: TODO refuse to migrate from other cnis

**Comment by emosbaugh** [comment]
[sc-54473]

**Review by laverya — APPROVED** [review]

**Pull request merged**